abci/types/types.proto

```diff
@@ -256,26 +256,24 @@ message Header {
   string chain_id = 2 [(gogoproto.customname)="ChainID"];
   int64 height = 3;
   google.protobuf.Timestamp time = 4 [(gogoproto.nullable)=false, (gogoproto.stdtime)=true];
-  int64 num_txs = 5;
-  int64 total_txs = 6;
 
   // prev block info
-  BlockID last_block_id = 7 [(gogoproto.nullable)=false];
+  BlockID last_block_id = 5 [(gogoproto.nullable)=false];
```

Comment:
Will bring them back for compatibility purposes

Reply:
This will be breaking anyway, so why not do this?!

Reply:
Because we're talking about breaking different things. The original change is breaking on the blockchain, but not necessarily on any APIs. So all you have to do is start a new blockchain to take advantage of the changes. But with a change like this, it's breaking on the ABCI, which means everyone has to update their ABCI stuff.

I don't think we should take a "breaking is breaking" attitude because we have like 6 different interfaces that can break and breaking one does not mean we should break another because each one has different implications for users. So the analysis should be based on the implications of a particular break and always be case specific, not just "breaking is breaking".

This is especially true here since we plan on breaking the header again to remove more stuff. I'd rather this change be non-breaking on the ABCI if possible, and a later change when we do more cleanup be breaking. At least then it might be worth it (a significantly simpler Header structure) vs just removing two fields now ...

Reply:
Our philosophy on breaking needs to be documented and committed to the repo somewhere ...

Reply:
Ok I'm more sympathetic to this because otherwise we'd have to annotate the `types#Header` to match the non-breaking removal of the fields and that would be painful and I'm not even sure amino supports it. In any case, the thoughts about "breaking is breaking" still stand :)

Reply:
@tessr Do you want to try to write such a document? It may help you get deeper understanding of various things that are breaking ("6 different interfaces")